collectors/python.d.plugin/samba/samba.chart.py

```diff
@@ -107,25 +108,27 @@ def __init__(self, configuration=None, name=None):
         self.rgx_smb2 = re.compile(r'(smb2_[^:]+|syscall_.*file_bytes):\s+(\d+)')
 
     def check(self):
-        sudo_binary = find_binary(SUDO)
-        if not sudo_binary:
-            self.error("can't locate '{0}' binary".format(SUDO))
-            return False
-
         smbstatus_binary = find_binary(SMBSTATUS)
         if not smbstatus_binary:
             self.error("can't locate '{0}' binary".format(SMBSTATUS))
             return False
 
-        command = [sudo_binary, '-n', '-l', smbstatus_binary, '-P']
-        smbstatus = '{0} -P'.format(smbstatus_binary)
-        allowed = self._get_raw_data(command=command)
-        if not (allowed and allowed[0].strip() == smbstatus):
-            self.error("not allowed to run sudo for command '{0}'".format(smbstatus))
-            return False
-
-        self.command = ' '.join([sudo_binary, '-n', smbstatus_binary, '-P'])
-        return ExecutableService.check(self)
+        if os.getuid() != 0:
+            sudo_binary = find_binary(SUDO)
+            if not sudo_binary:
+                self.error("can't locate '{0}' binary".format(SUDO))
+                return False
+            command = [sudo_binary, '-n', '-l', smbstatus_binary, '-P']
+            smbstatus = '{0} -P'.format(smbstatus_binary)
+            allowed = self._get_raw_data(command=command)
+            if not (allowed and allowed[0].strip() == smbstatus):
+                self.error("not allowed to run sudo for command '{0}'".format(smbstatus))
+                return False
+            self.command = ' '.join([sudo_binary, '-n', smbstatus_binary, '-P'])
+            return ExecutableService.check(self)
+        else:
+            self.command = ' '.join([smbstatus_binary, '-P'])
+            return ExecutableService.check(self)
```

Comment:
i suggest to not use `else` 😄 it is easier to read imo 

```python
if os.getuid() == 0:
  ...

...
```

Reply:
This should be resolved now :)